tests/test_core/test_events.py

```diff
@@ -353,3 +353,116 @@ def test_rotate(
+    example_events_data_frameclock,
+) -> None:
+
+    # Arrange
+    data_minimal = Events(example_events_data_minimal)
+    data_none = Events(example_events_data_minimal_none)
+    data_frameclock = Events(example_events_data_frameclock)
+
+    # Act
+    data_minimal_full = data_minimal.slice(inplace=False)
+    data_minimal_full_arguments = data_minimal.slice(start=1.1, end=2.3, inplace=False)
+    data_minimal_end_sliced = data_minimal.slice(end=2.2, inplace=False)
+    data_minimal_start_sliced = data_minimal.slice(start=1.2, inplace=False)
+    data_none_sliced = data_none.slice(inplace=False)
+    data_frameclock_sliced = data_frameclock.slice(
+        end=15, slice_by="frameclock", inplace=False
+    )
+
+    # Assert
+    assert data_minimal_full.events.shape == (2, 2)
+    assert data_minimal_full_arguments.events.shape == (2, 2)
+    assert data_minimal_end_sliced.events.shape == (1, 2)
+    assert data_minimal_start_sliced.events.shape == (1, 2)
+    assert data_none_sliced.events.shape == (1, 2)
+    assert data_frameclock_sliced.events.shape == (1, 3)
+
+
+@pytest.mark.unit
+def test_slice_inplace(
+    example_events_data_minimal,
+    example_events_data_minimal_none,
+    example_events_data_frameclock,
+) -> None:
+
+    # Arrange
+    data_minimal = Events(example_events_data_minimal)
+    data_none = Events(example_events_data_minimal_none)
+    data_frameclock = Events(example_events_data_frameclock)
+
+    # Act
+    data_minimal.slice(start=10, inplace=True)
+    data_none.slice(inplace=True)
+    data_frameclock.slice(start=15, slice_by="frameclock", inplace=True)
+
+    # Assert
+    assert data_minimal.events.shape == (0, 2)
+    assert data_none.events.shape == (1, 2)
+    assert data_frameclock.events.shape == (1, 3)
+
+
+@pytest.mark.unit
+def test_get_event_stream_frameclock(
+    example_events_data_minimal,
+    example_events_data_frameclock,
+):
+    # Arrange
+    data_minimal = Events(example_events_data_minimal)
+    data_frameclock = Events(example_events_data_frameclock)
+
+    # Act
+    try:
+        event_stream_minimal = data_minimal.get_event_stream()
+    except ValueError:
+        event_stream_minimal = None
+    event_stream_frameclock = data_frameclock.get_event_stream()
+
+    # Assert
+    assert event_stream_minimal is None
+    assert len(event_stream_frameclock) == 6
+    assert event_stream_frameclock[0] == 1
+    assert event_stream_frameclock[-1] == 2
+    assert np.nansum(event_stream_frameclock[1:5]) == 0
+
+
+@pytest.mark.unit
+def test_get_event_stream_frameclock_none(
+    example_events_data_frameclock_none,
+):
+    # Arrange
+    data_frameclock_none = Events(example_events_data_frameclock_none)
+
+    # Act
+    event_stream_frameclock_none = data_frameclock_none.get_event_stream()
+
+    # Assert
+    assert len(event_stream_frameclock_none) == 1
+    assert event_stream_frameclock_none[0] == "2"
+
+
+@pytest.mark.unit
+def test_get_event_stream_frameclock_unsorted(
```

Comment:
just short inline-comments explaining the test would help (e.g. what is unsorted?)